### DISCUSSION TITLE
Water Currents Causing Performace Issues

### DISCUSSION THREAD

**Discussion by 4-S-H** [FakeFishGames/Barotrauma#15437] [Ideas and suggestions]
It appears that the game performance has improved notably after the levelobjects watercurrent1 and watercurrentvertical were overriden to empty.

**Comment by SomeRandomNoobKekeke** [comment]
Ye, obviously, they have triggers pushing things around, and also they update their force every frame

You can also cut wrecks or ruins from the game and they won't lag, or water particles if you have weak gpu

The quesion is: is there an easy way to fix it without removing it

**Comment by SomeRandomNoobKekeke** [comment]
I seems that "in rest" water currents are no laggier than cave roots/vents
![20241225071941_1](https://github.com/user-attachments/assets/4c928a2e-7fd2-4b3c-b54c-7394752e1514)

But when there's a lot of bodies floating around they're taking the lead 
![20241225072047_1](https://github.com/user-attachments/assets/24186202-df0a-4102-b4e1-07d87c7e0a3c)

[There's a solution at least for dead bodies](https://steamcommunity.com/sharedfiles/filedetails/?id=3222992798)
